core/src/main/scala/kafka/server/ReplicaAlterLogDirsThread.scala

```diff
@@ -76,7 +76,8 @@ class ReplicaAlterLogDirsThread(name: String,
     var partitionData: Seq[(TopicPartition, FetchData)] = null
     val request = fetchRequest.build()
 
-    val (topicIds, topicNames) = replicaMgr.metadataCache.topicIdInfo()
+    val topicIds = request.data().topics().asScala.map(topic => (topic.topic, topic.topicId)).toMap
+    val topicNames = topicIds.map(_.swap)
```

Comment:
1) Is it fine to rely on the request to construct the topic to topicId mapping? I suppose that it is OK here because the request was populated with both, right? If this is correct, is it worth adding a small comment here?

2) nit: We could omit few parenthesis here (e.g. `data`, `topics`).

3) Is it worth using two mutable Maps which would be populated while iterating over the `topics`?

Reply:
1. Yes, the LeaderAndIsr request contains both name and ID
2. will do
3. that makes sense -- i can also set the map size early on so resizing won't be an issue.

Reply:
Ah looks like I can't set the default for earlier scala versions.